src/AccountServices.java

```diff
@@ -54,185 +77,151 @@ public String register(String outfile) throws IOException {
             PIN = scanner.nextLine();
         }
 
-        Files.write(Paths.get(outfile), (userName+"\t"+ IBAN+"\t"+ PIN).concat("\n").getBytes(), StandardOpenOption.APPEND);
+        Document doc = new Document("userName",userName)
+                .append("IBAN", IBAN)
+                .append("PIN", PIN);
+        credentials.insertOne(doc);
 
         System.out.println();
         System.out.println(ANSI_GREEN + "Your account has been created." + ANSI_RESET);
         System.out.println();
         return userName;
     }
 
-    public double getCurrentBalance(String userName) throws IOException {
-        double currentBalance = 0;
-        File file = new File("TransactionHistory.txt");
-        FileReader f = new FileReader(file);
-        BufferedReader b = new BufferedReader(f);
+    public double getCurrentBalance(String userName, MongoCollection <Document> transactions) {
+        final double[] currentBalance = {0};
 
-        String line;
-        while((line = b.readLine()) != null) {
-            String[] columns = line.split("\t");
+        Block<Document> balanceBlock = new Block<Document>() {
+            @Override
+            public void apply(final Document document) {
+                ObjectMapper objectMapper = new ObjectMapper();
+                String jsonHistory = document.toJson();
+                JsonNode jsonNode = null;
 
-            if(columns[0].toUpperCase().equals(userName.toUpperCase()) )
-            {
-                switch (columns[2])
+                try {
+                    jsonNode = objectMapper.readTree(jsonHistory);
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+
+                switch (jsonNode.get("SERVICE").asText())
                 {
-                    case "WITHDRAW" : {
-                        currentBalance -= Double.parseDouble(columns[4]);
-                        break;
-                    }
+                    case "WITHDRAW" :
 
-                    case "DEPOSIT" : {
-                        currentBalance += Double.parseDouble(columns[4]);
+                    case "BILLING" : {
+                        currentBalance[0] -= Double.parseDouble(jsonNode.get("AMOUNT").asText());
```

Comment:
I needed to use currentBalance variable within the switch and because it had to be initialized to 0 before the switch, it had to be final. However, a final variable which is initialized, cannot receive another value along the program, and I had to make it a one value array in order to do so.